exporter/exporter_test.go

```diff
@@ -421,3 +450,61 @@ func TestExporterStartNoMetricsSelected(t *testing.T) {
 		defer exp.Stop()
 	}
 }
+
+func testBasicAuth(t *testing.T, opts *NATSExporterOptions, testuser, testpass string, expectedRc int) error {
+	exp := NewExporter(opts)
+	if err := exp.Start(); err != nil {
+		return err
+	}
+	defer exp.Stop()
+
+	return checkExporterFull(t, testuser, testpass, exp.http.Addr().String(), false, expectedRc)
+}
+
+func TestExporterBasicAuth(t *testing.T) {
+	opts := getDefaultExporterTestOptions()
+	opts.ListenAddress = "localhost"
+	opts.ListenPort = 0
+	opts.GetVarz = true
+	opts.GetConnz = true
+	opts.GetSubz = true
+	opts.GetRoutez = true
+	opts.HTTPUser = "colin"
+	opts.HTTPPassword = "password"
+
+	s := pet.RunServer()
+	defer s.Shutdown()
+
+	// first try user/pass with no auth.
+	testBasicAuth(t, opts, "colin", "password", http.StatusOK)
+
+	// now try user/pass
+	opts.HTTPUser = "colin"
+	opts.HTTPPassword = "password"
```

Comment:
I don't understand this. The original `opts` has already set the user/password and we already tried `testBasicAuth()` 3 lines above.

Reply:
It's a bug in the test...  thanks!